src/main/java/com/uid2/shared/model/Site.java

```diff
@@ -38,107 +40,40 @@ public Site(@JsonProperty("id") int id,
-        this.enabled = enabled;
-        this.clientTypes = (types != null) ? new HashSet<>(types) : new HashSet<>();
-        this.domainNames = (domains != null) ? new HashSet<>(domains) : new HashSet<>();
-        this.visible = visible;
-        this.created = Instant.now().getEpochSecond();
+        this(id, name, description, enabled, types, domains, new HashSet<>(), visible);
     }
 
-    public int getId() { return id; }
-    public String getName() { return name; }
-    public String getDescription() { return description; }
-    public Boolean isEnabled() { return enabled; }
-    public Set<String> getDomainNames() { return domainNames; }
-    public Set<ClientType> getClientTypes() { return clientTypes; }
-    public Boolean isVisible() { return visible; }
-
-    public long getCreated() { return  created; }
-
-    public void setDescription(String description) { this.description = description; }
-
-    public void setClientTypes(Set<ClientType> clientTypes) { this.clientTypes = clientTypes; }
-
-    public void setEnabled(Boolean enabled) { this.enabled = enabled; }
-    public void setDomainNames(Set<String> domainNames) { this.domainNames = domainNames; }
-    public void setVisible(Boolean visible) { this.visible = visible; }
-
-    @Override
-    public String toString() {
-        return "Site{" +
-                "id=" + id +
-                ", name='" + name + '\'' +
-                ", description=" + description +
-                ", enabled=" + enabled +
-                ", domain_names=" + domainNames.toString() +
-                ", clientTypes=" + clientTypes.toString() +
-                ", visible=" + visible +
-                ", created=" + String.valueOf(created) +
-                '}';
+    public Site(int id, String name, Boolean enabled, Set<ClientType> types, Set<String> domains, Set<String> appNames) {
+        this(id, name, DEFAULT_DESCRIPTION, enabled, types, domains, appNames, DEFAULT_VISIBLE);
     }
 
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-        Site site = (Site) o;
-        return id == site.id && name.equals(site.name) && description.equals(site.description) && enabled.equals(site.enabled) && domainNames.equals(site.domainNames) && clientTypes.equals(site.clientTypes) && Objects.equals(visible, site.visible) && created == site.created;
+    public Site(int id, String name, String description, Boolean enabled, Set<ClientType> types, Set<String> domains, Set<String> appNames, Boolean visible) {
+        this(id, name, description, enabled, types, domains, appNames, visible, Instant.now().getEpochSecond());
     }
 
-    @Override
-    public int hashCode() {
-        return Objects.hash(id, name, description, enabled, domainNames,Arrays.hashCode(clientTypes.toArray()), visible, created);
-    }
+    public Boolean isEnabled() { return enabled; }
```

Comment:
these getter methods can be generated by lombok, right ?

Reply:
Lombok generated the method `getEnabled()` but we use `isEnabled()`

Reply:
Oh, ok. Its probably because its a boxed type instead of primitive boolean. In the ctor a default value is always assigned so I don't see a reason for it to keep it boxed. Nitpicking here, I'll leave it upto you :)

Reply:
I will just leave as it is now :)